nixos/modules/security/opendoas.nix

```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.security.opendoas;
+
+  inherit (pkgs) opendoas;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    security.opendoas.enable = mkOption {
+      type = types.bool;
+      default = true;
+      description =
+        ''
+          Whether to enable the <command>doas</command> command, which
+          allows non-root users to execute commands as root.
+        '';
+    };
+
+    security.opendoas.configFile = mkOption {
+      type = types.lines;
+      # Note: if syntax errors are detected in this file, the NixOS
+      # configuration will fail to build.
+      description =
+        ''
+          This string contains the contents of the
+          <filename>doas.conf</filename> file.
+        '';
+    };
+
+    security.opendoas.extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Extra configuration text appended to <filename>doas.conf</filename>.
+      '';
+    };
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    security.opendoas.configFile =
```

Comment:
Hm, looks to me like the `configFile` config value is simply discarded. I'd instead expect the option to be written to `/etc/doas.conf`. As is, I don't understand how `doas` is supposed to get at the configuration.